### DISCUSSION TITLE
Fullscreen over several monitors

### DISCUSSION THREAD

**Discussion by strandenzee** [pop-os/cosmic-epoch#1379] [Ideas]
I often use a remote session where i need a two monitor setup. It would be nice to have the option to get a full-screen over multiple monitors. Maybe one workspace over two monitors where the window can be set to full-screen. It is more then just stretching the window over two monitors, but rather a full-screen behaving like two monitors. They can have different setups. In my case one of these two screens is rotated 90 deg.
In i3 there is an option for global full-screen, but I would like to have an option of not using all available  screens used by this type of full-screen mode. More like monitor 1 and 2 acting for one full-screen and monitor 3 and 4 like normal tiling windows.

**Comment by Drakulix** [comment]
That is difficult to provide on the DE side of things. The reason this works in i3, is that X11 lays out screens on a global coordinate grid. Something COSMIC doesn't do, so this request is somewhat incompatible with our current architecture.

It would be much more straight-forward to ask your client app to create separate windows for separated outputs, which then could be made fullscreen separately on whichever output you'd choose.

**Comment by yagler** [comment]
I have played with and really enjoy Cosmic's tiling. The main use of my Linux desktop is to work from home.
I have 2 monitor's at home and two in the office. In KDE-Wayland I'm able to install the Horizon client (or use the web version) and have it treat both displays separately and work with full screen on both of them. In Cosmic I can't get an application to full screen on multiple displays or even stretch across multiple displays (at which point I could try and tile my remote desktop so that it acted a bit like it was two screens). @Drakulix, when you mention being incompatible with the current architecture does that mean we can't stretch an application across multiple displays at all?

Thanks for everyone's work on Cosmic it is looking great and could be my daily driver if this multi-monitor remote session was do-able.

**Reply by Stolenskiy** [reply]
I would also like to emphasize that this feature is very important to me, similar to what @yagler mentioned.

My main use case is a remote desktop session through Omnissa Horizon Client. On my workstation I use two monitors, and in other desktop environments / operating systems I can run the remote Windows session fullscreen across both displays.

In COSMIC, the client window can become wide enough for two monitors, but it does not behave like a real fullscreen surface spanning both displays. It looks like the second desktop/workspace remains above or clips the remote desktop window, so the remote session is not usable in a proper dual-monitor fullscreen mode.

For my workflow this is not just a cosmetic issue. Dual-monitor fullscreen remote desktop is essential for daily work, especially when working with enterprise VDI environments. Without this, I either have to use only one monitor or switch to another desktop environment/session specifically for remote work.

I understand that COSMIC’s model is different from the traditional X11 global coordinate space, but support for fullscreen windows across multiple monitors, or an alternative mechanism that allows remote desktop clients to use multiple outputs correctly, would be extremely valuable.
